### PR TITLE
Streaming fix

### PR DESCRIPTION
**Pull request by ifrit98**
* Separate out streaming from non-streaming requests (`AsyncGenerator` return value)
* Yield chunks as they are sent

### CODE REVIEW

File: bittensor/dendrite.py [hunk 1]
```diff
@@ -133,66 +133,94 @@ async def forward(
...
-                    )
-                # The function then returns a list of responses from all axons.
-                return all_responses
-            else:
-                # Here we build a list of coroutines without awaiting them.
-                coroutines = [
-                    self.call(
+        # Check if synapse is an instance of the StreamingSynapse class or if streaming flag is set.
+        streaming = (
+            issubclass(synapse.__class__, bittensor.StreamingSynapse) or streaming
```
Comment (Eugene-hu) [inline bittensor/dendrite.py]: Should there be a warning if the two don't match? Should we expect the base class to be streaming if streaming is used?

**Review by Eugene-hu — APPROVED** [review]
LGTM, left you a comment

File: bittensor/dendrite.py (hunk 1 above)
Comment (ifrit98) [reply bittensor/dendrite.py]: Yes, we should expect to subclass, but if subnet owners decide that the `bt.StreamingSynapse` is too restrictive for their purposes, they should be able to write their own class that inherits from `bt.Synapse`.

Comment (ifrit98) [reply bittensor/dendrite.py]: Agree we should add a warning here.

**Pull request merged**